File: modules/common-model/src/main/java/com/shterneregen/securelan/common/model/ConnectionInfo.java
package com.shterneregen.securelan.common.model;

import java.util.Objects;

public record ConnectionInfo(
        String host,
        int port,
        boolean secure
) {
    public ConnectionInfo {
        Objects.requireNonNull(host, "host must not be null");
        if (port < 1 || port > 65535) {
            throw new IllegalArgumentException("port must be between 1 and 65535");
        }
    }
}

File: modules/common-model/src/main/kotlin/com/shterneregen/securelan/common/model/ConnectionInfo.kt
package com.shterneregen.securelan.common.model

import java.util.Objects

@JvmRecord
data class ConnectionInfo(
    val host: String,
    val port: Int,
    val secure: Boolean,
) {
    init {
        Objects.requireNonNull(host, "host must not be null")
        require(port in 1..65_535) { "port must be between 1 and 65535" }
    }
}
